src/erldns_packet_cache.erl

```diff
@@ -19,133 +19,91 @@ same question received within the cache TTL.
 
 The cache is swept for old cache data at regular intervals.
 
-Emits the following telemetry events:
-- `[erldns, cache, expired]`
-- `[erldns, cache, miss]`
-- `[erldns, cache, hit]`
+## Configuration
+
+```erlang
+{erldns, [
+    {packet_cache, #{
+        enabled := boolean(), %% defaults to true
+        ttl := non_neg_integer(), %% Seconds, defaults to 30
+    }}
+]}
+```
+
+## Telemetry events
+
+See `m:segmented_cache` for telemetry events. The name is `erldns_cache`.
 """.
 
--behaviour(gen_server).
+-include_lib("dns_erlang/include/dns.hrl").
 
-% API
 -export([
     start_link/0,
     get/1,
     get/2,
     put/2,
-    sweep/0,
-    clear/0,
-    stop/0
-]).
-% Gen server hooks
--export([
-    init/1,
-    handle_call/3,
-    handle_cast/2,
-    handle_info/2,
-    terminate/2,
-    code_change/3
+    clear/0
 ]).
 
--define(SERVER, ?MODULE).
+-define(NAME, erldns_cache).
```

Comment:
From a naming perspective, `erldns_cache` is pretty general. I find the clarity of `erldns_packet_cache` a bit better as it is clearer on what is in the cache (versus a zone cache, or an rr_set cache for example).

Reply:
Tru 👌🏽